### PR TITLE
Tests result save

### PR DESCRIPTION
**Pull request by harnash**
Add saving TestRun results on Task completition. Added Auto Discovery of API URI's. Fixed some Python 2.x/3.x compatibility issues.

### CODE REVIEW

File: apps/testrunner/runner.py [hunk 1]
```diff
@@ -5,22 +5,47 @@
...
 import six
 
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
 from .logger import logger
 from .test_suites import SimpleTestSuite
 
 
+class AutoDiscoverFailed(ImproperlyConfigured):
```
Comment (wladekb) [inline apps/testrunner/runner.py]: "AutoDiscoveryFailed" ?

File: apps/testrunner/runner.py [hunk 2]
```diff
@@ -5,22 +5,47 @@
...
+            api_descr = req.json()
+
+            self.TASKS_API_URL = api_descr['tasks']
+            self.RESULTS_API_URL = api_descr['results']
+        except Exception as ex:
+            raise AutoDiscoverFailed('Cannot find API URIs: {0}'.format(api_descr)) from ex
 
     def acquire(self):
-        url = '{}/fetch/'.format(self.URL)
+        url = '{}fetch'.format(self.TASKS_API_URL)
```
Comment (wladekb) [inline apps/testrunner/runner.py]: The trailing slash was needed for some reason when I tested it due to some configuration setting? Please make sure to double-check if it works correctly and does not do a redirect.

File: apps/testrunner/ssh.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import os
 import socket
 import tempfile
-from six import StringIO
+from six import BytesIO
```
Comment (wladekb) [inline apps/testrunner/ssh.py]: I guess BytesIO are probably better here but could you share what is the difference in this specific situation?

File: apps/testrunner/runner.py (hunk 2 above)
Comment (harnash) [inline apps/testrunner/runner.py]: Yes I know :) I did unified those URIs so they are consistent throughout the app. API returns always with trailing slash (unless configured differently?). It does work now.

Comment (harnash) [inline apps/testrunner/runner.py]: Right. It works but trailing slash should be there: bd4647a3dc1a53b889298007fb5eeb10a0935bef

File: apps/testrunner/ssh.py (hunk 3 above)
Comment (harnash) [inline apps/testrunner/ssh.py]: The only difference is that it works for Python 2.x and Python 3.x when you deal with unicode strings. StringIO doesn't work well in Pythin 3.x and data received from channel.

**Pull request merged**